### ISSUE TITLE
Improve consistency between styling of React Bootstrap and React Overlays docs

### ISSUE THREAD

**Issue by bpas247** [react-bootstrap/react-overlays#584]
## Is your feature request related to a problem? Please describe

Currently, the styling between the React Bootstrap and React Overlays docs are *way* off. We should aim to improve the consistencies between our docs, so that our end users won't get confused when they navigate between our docs sites 

## Describe the solution you'd like

Improve the styling between the React Bootstrap and React Overlays docs. This could include changes such as:

- improved consistency of styling
- Add a Navbar (similar to the one implemented for the React Bootstrap docs, but with less links)
- Add a Searchbar for the left side panel.
- Add a right side panel (to show where the user currently is while scrolling through that section of the docs).

## Describe alternatives you've considered

There's a few other things we could include here, such as:
- Add a "Getting Started" section.
- Add a "Why React Overlays" section.

**Comment by Nikhil-Vats** [comment]
Hey, @bpas247 . I would love to contribute to this. I believe this is a big issue and could be broken down to smaller ones. 

- Correct me if I am wrong but you want a navbar on [react-overlay-docs](https://react-bootstrap.github.io/react-overlays/modal/) just like the one [here](https://react-bootstrap.github.io/).

-  You also want a searchbar and right side panel.

I will send a PR and update for one component at a time and update it regularly. I will send the PR for navbar by tonight.

**Comment by Nikhil-Vats** [comment]
I also noticed that if I go to this URL - https://react-bootstrap.github.io/react-overlays/. There is just a blank page. I think it would be better if we either redirect it to  https://react-bootstrap.github.io/react-overlays/modal/ or design a new homepage. Your views?

**Comment by bpas247** [comment]
> Hey, @bpas247 . I would love to contribute to this. I believe this is a big issue and could be broken down to smaller ones.

Yeah there's a lot of moving parts for this one, I agree that it would be more easy if this was just broken up into smaller issues. I think I might end up closing this issue after a discussion around the scope has finalized what we want to do for this.

> Correct me if I am wrong but you want a navbar on react-overlays-docs just like the one here.

Yep! I figured that would make the most sense consistency-wise if we kept a similar design for the navbar.

> You also want a searchbar and right side panel.

I think the searchbar will probably be low priority for now, but a right side panel similar to the one implemented for the [React Bootstrap website](https://react-bootstrap.github.io/getting-started/introduction#installation) would make it much easier for the user to tell where exactly they are on the page.

> I also noticed that if I go to this URL - https://react-bootstrap.github.io/react-overlays/. There is just a blank page. I think it would be better if we either redirect it to https://react-bootstrap.github.io/react-overlays/modal/ or design a new homepage. Your views?

That looks like a bug tbh, I think we should push a hotfix to re-direct the root route to the modal route for now and then have some more discussion for a redesign for a new homepage later.
